### ISSUE TITLE
Diamond spikes killing specific Infernal Mobs causes crash

### ISSUE THREAD

**Issue by FakoTheGreat** [rwtema/extrautilities_old_issues#489]
If an Infernal Mob would normally apply a debuff to a player (such as Darkness or Weakness mobs), a diamond spike attempting to kill that mob will crash the game.  The mob is not killed prior to the crash, meaning that an external editor is required to remove the mob or the block in order to recover the save.  The crash can be replicated in SSP or SMP, and does not occur with any lower grade of spike.

Relevant versions:
Forge 1236
Extra Utilities 1.2.1
Infernal Mobs 1.7.10

Crash log:
https://gist.github.com/FakoTheGreat/add20e28d43e9a061836

**Comment by FreneticScribbler** [comment]
Probably related to whatever way Tema convinces the mob it was killed by a player so it can drop "player-specific drops", maybe a FakePlayer. This could probably be fixed by either Tema or Infernal Mobs author, probably best to take it to author of Infernal Mobs, they just need to check the player isn't null before applying effects.

**Comment by TherminatorX** [comment]
Nah, this is likely producible by hitting one with a TE Autonomous Activator in left-click mode. As this is a crash with mobs that applies effects to what hits it, I'm able to guess they're trying to apply those effects to the block.

Test it by making a AA hit a IM mob with those attributes. If it crashes, then we know it's only IM's fault. If not, we know there's a coding conflict between XU and IM.

**Comment by Chromotron** [comment]
A very similiar crash happens if one uses diamond spikes to kill Thaumcraft Giant Taintacles (or probably any other TC boss).
